sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java

```diff
@@ -367,10 +374,17 @@ private synchronized void checkPushInIntent(Activity activity, Intent intent) {
                 if (mFirstGoogleMessageId == null) {
                     mFirstGoogleMessageId = googleMessageId;
                 }
-                PushNotification notification = new PushNotification(intent);
+
                 AppCenterLog.info(LOG_TAG, "Clicked push message from background id=" + googleMessageId);
                 mLastGoogleMessageId = googleMessageId;
                 AppCenterLog.debug(LOG_TAG, "Push intent extras=" + intent.getExtras());
+
+                /* Try to get the received push intent via google message id. */
+                Intent currentPushIntent = mReceivedIntents.get(googleMessageId);
+                if (currentPushIntent == null) {
```

Comment:
This is precisely what will make the feature unreliable. If the process is restarted before clicking on the notification will be missing title and message. This is thus not a solution that should get recommended as users will need guarantee if they need to read those fields.